flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/statistics/basicstatistic/BaseSummarizer.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.common.statistics.basicstatistic;
+
+import org.apache.flink.ml.common.linalg.DenseMatrix;
+
+import java.io.Serializable;
+
+/**
+ * Summarizer is the base class to calculate summary and store intermediate results, and Summary is the result of Summarizer.
```

Comment:
I actually have a higher level questions regarding the type of summary we are going to support and what's the road map in the future.

In general: a "summary" to me should be something that can be serialized into a format that an external system can read and use it to produce meaningful insights. such as visualization e.g. [tf.summary](https://www.tensorflow.org/api_docs/python/tf/summary), or for transforming into some object that can be used to export data [R.summary](https://www.rdocumentation.org/packages/base/versions/3.6.1/topics/summary). 

In this case, the class "summary" in this PR is more like a wrapper around: `sum()`, `count()`, `correlate()`, `covariance()`. I am not sure what's the value add to the ML-library. Maybe I was understanding this entire intention in the wrong way.

Reply:
To add to this discussion. Some of the summaries used by Tensorflow for example, is meant to be used by Tensorboard - which understands the format of the summary in order to plot visualization via [protobuf](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/summary.proto)), same thing with [tf.Event](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/util/event.proto). 

If this is the case, is there an existing discussion regarding what needs to be done in terms of the public APIs of a `Summary` class should have in Flink-ML? I haven't follow up closely with the ML but I cant find any in the [FLIP-39 documentation](https://cwiki.apache.org/confluence/display/FLINK/FLIP-39+Flink+ML+pipeline+and+ML+libs)

Reply:
Tranks for your comments @, We just provide a simple way to view the statistics, similar to summarizer in SparkML. The value can be printed. For example,
```
Summarizer summarizer = new Summarizer(data)
  .setSelectedCols("f_double", "f_int");
TableSummary srt = summarizer.collectResult();
System.out.println(srt.mean("f_double”));
```

Summarizer is also a tool, which can be used by other algorithms.  Call mean () and standarddeviation () for the standardized data of logical regression.